File: l10n_gr_partner_pos/static/src/models/res_partner.js
/** @odoo-module **/

import { patch } from "@web/core/utils/patch";
import { ResPartner } from "@point_of_sale/app/models/res_partner";

patch(ResPartner.prototype, {
    get searchString() {
        const base = super.searchString;
        const extras = ["eponymia", "kinito", "doy", "drastiriotita"]
            .map((f) => {
                const v = this[f];
                if (!v) return "";
                if (f === "kinito") {
                    return v.replace(/[+\s()-]/g, "");
                }
                return v;
            })
            .filter(Boolean)
            .join(" ");
        return extras ? `${base} ${extras}` : base;
    },
});

File: l10n_gr_partner_pos/models/res_partner.py
# -*- coding: utf-8 -*-
from odoo import api, models


class ResPartner(models.Model):
    _inherit = "res.partner"

    @api.model
    def _load_pos_data_fields(self, config):
        fields = super()._load_pos_data_fields(config)
        for f in ("eponymia", "kinito", "doy", "drastiriotita"):
            if f not in fields:
                fields.append(f)
        return fields
